tests/tests/test-democracy.ts

```diff
@@ -0,0 +1,301 @@
+
+describeDevMoonbeam("Democracy - forget notePreimage", (context) => {
+  let genesisAccount: KeyringPair, alith: KeyringPair;
+  let encodedHash: string;
+  let enactmentPeriod, votingPeriod;
+
+  before("Setup genesis account for substrate", async () => {
+    const keyring = new Keyring({ type: "ethereum" });
+    genesisAccount = await keyring.addFromUri(GENESIS_ACCOUNT_PRIVATE_KEY, null, "ethereum");
+    alith = await keyring.addFromUri(ALITH_PRIV_KEY, null, "ethereum");
+
+    // notePreimage
+    // compute proposal hash but don't submit it
+    const encodedProposal =
+      context.polkadotApi.tx.parachainStaking
+        .setParachainBondAccount(GENESIS_ACCOUNT)
+        .method.toHex() || "";
+    encodedHash = blake2AsHex(encodedProposal);
+  });
+  it("vote", async function () {
+    this.timeout(200000);
+
+    // propose
+    const { events: eventsPropose } = await createBlockWithExtrinsic(
+      context,
+      genesisAccount,
+      context.polkadotApi.tx.democracy.propose(encodedHash, PROPOSAL_AMOUNT)
+    );
+    expect(eventsPropose[5].toHuman().method).to.eq("ExtrinsicSuccess");
+    await context.createBlock();
+    // second
+    const { events: eventsSecond } = await createBlockWithExtrinsic(
+      context,
+      alith,
+      context.polkadotApi.tx.democracy.second(0, 1000)
+    );
+    expect(eventsSecond[2].toHuman().method).to.eq("ExtrinsicSuccess");
+    // let Launchperiod elapse to turn the proposal into a referendum
+    // launchPeriod minus the 3 blocks that already elapsed
+    for (let i = 0; i < 7200; i++) {
+      await context.createBlock();
+    }
+    // referendumCount
+    let referendumCount = await context.polkadotApi.query.democracy.referendumCount();
+    expect(referendumCount.toHuman()).to.equal("1");
+
+    // vote
+    await context.createBlock();
+    const { events: eventsVote } = await createBlockWithExtrinsic(
+      context,
+      alith,
+      context.polkadotApi.tx.democracy.vote(0, {
+        Standard: { balance: VOTE_AMOUNT, vote: { aye: true, conviction: 1 } },
+      })
+    );
+    expect(eventsVote[1].toHuman().method).to.eq("ExtrinsicSuccess");
+
+    // referendumInfoOf
+    const referendumInfoOf = await context.polkadotApi.query.democracy.referendumInfoOf(0);
+    expect((referendumInfoOf.toHuman() as any).Ongoing.proposalHash).to.equal(encodedHash);
+    expect((referendumInfoOf.toHuman() as any).Ongoing.tally.ayes).to.equal("10.0000 UNIT");
+    expect((referendumInfoOf.toHuman() as any).Ongoing.tally.turnout).to.equal("10.0000 UNIT");
+
+    // let votePeriod + enactmentPeriod elapse to turn the proposal into a referendum
+    for (let i = 0; i < Number(votingPeriod) + Number(enactmentPeriod); i++) {
+      await context.createBlock();
+    }
+    // the enactement should fail
+    let parachainBondInfo = await context.polkadotApi.query.parachainStaking.parachainBondInfo();
+    expect(parachainBondInfo.toHuman()["account"]).to.equal(ZERO_ADDRESS);
```

Comment:
You could assert that it is not still in the dispatch queue

Reply:
How do I test that? How do I access dispatch queue? @JoshOrndorff

Reply:
Here are the storage items in pallet scheduler that you could look in. To me it seems like `Lookup` would be easier, but up to you. So I guess something like `api.query.scheduler.lookup(task_id).is_some()`. https://github.com/paritytech/substrate/blob/master/frame/scheduler/src/lib.rs#L172-L185

The task_id that democracy uses is `(DEMOCRACY_ID, index)`. https://github.com/paritytech/substrate/blob/master/frame/democracy/src/lib.rs#L1706

Reply:
Also, no need to hold this whole PR up on this, it was just a minor suggestion. If it turns out to be hard, I'm fine merging this as it is (already approved).

Reply:
Created a ticket in JIRA